plugins/modem-manager/fu-mm-dfota-device.c

```diff
@@ -431,7 +427,12 @@ fu_mm_dfota_device_write_firmware(FuDevice *device,
 		g_prefix_error(error, "failed to start update: ");
 		return FALSE;
 	}
+
+	if (!fu_device_close(FU_DEVICE(self), error))
+		return FALSE;
 	fu_device_sleep(FU_DEVICE(self), FU_MM_DFOTA_DEVICE_FOTA_RESTART_TIMEOUT_SECS * 1000);
+	if (!fu_device_open(FU_DEVICE(self), error))
+		return FALSE;
```

Comment:
I don't understand this at all -- why are you closing and then re-opening the device like this? Could we move the delay to fu_mm_dfota_device_attach() and use the daemon provided close->open?

Reply:
Isn't close and open being called for the attach() action?

Reply:
I rebased this PR onto your commit.
Now there's only the switched up progress times left.

Reply:
> but unfortunately DFOTA devices don't properly replug directly

Do they emit anything as a uevent we can take as "a replug?

Reply:
After digging down further there _is_ a replug uevent, so `WAIT_FOR_REPLUG` should work in that case, but the machine with the modem does not support proper udev.
I'll try setting `HAVE_UDEV` to 0 and let libusb detect the replug.

Reply:
I've opened #8969 for the udev problems - Adding `fu_device_add_flag(device, FWUPD_DEVICE_FLAG_WAIT_FOR_REPLUG)` after `write_firmware` and `attach` should work, but I can't currently test it on my hardware (without compatible udev).